ts-client/src/amm/types/index.ts

```diff
@@ -112,22 +113,17 @@ export type DepegLido = {
 
 export type DepegType = DepegNone | DepegMarinade | DepegLido;
 
-export type Depeg = {
-  baseVirtualPrice: BN;
-  baseCacheUpdated: BN;
-  depegType: DepegType;
-};
-
 export interface TokenMultiplier {
   tokenAMultiplier: BN;
   tokenBMultiplier: BN;
   precisionFactor: number;
 }
 
-// PoolState
-export type PoolState = TypeDef<AmmIdl['accounts']['0'], IdlTypes<AmmIdl>>;
-export type VirtualPrice = TypeDef<AmmIdl['types']['4'], IdlTypes<AmmIdl>>;
-export type ApyState = TypeDef<AmmIdl['accounts']['1'], IdlTypes<AmmIdl>>;
+export type PoolState = Omit<IdlAccounts<AmmIdl>['pool'], 'curveType'> & { curveType: CurveType };
```

Comment:
<img width="670" alt="Screenshot 2023-03-03 at 12 58 14 PM" src="https://user-images.githubusercontent.com/99231275/222635176-5f2b3d10-f6c2-4919-b6ce-b22d390d4db5.png">

Looks like it's breaking `jupiter-core`, i think we need to union `PoolFees` as well

Reply:
How do you test this? i wanted to link using `rush-pnpm` but it looks convoluted to use

Reply:
U can get a temp version from here. https://github.com/mercurial-finance/mercurial-dynamic-amm-sdk/actions/runs/4314597163/jobs/7527841156#step:8:17

Reply:
quite interesting the previous typing made the whole top level type any, that is why we didn't have to do the retyping of inner types